src/views/RegisterForm/RegisterForm.jsx

```diff
@@ -0,0 +1,322 @@
+   */
+  const dismissAlert = () => {
+    setAlertNotification(defaultError);
+  };
+  /**
+   * Returns the value of the key given from the map.
+   * Map value is the result of the regex.
+   * Object that returns the proper valid input function.
+   * Lots of if statements.
+   * @param {String} name - the input name.
+   * @param {String} value - the value of the input field.
+   * */
+  const getValidationFunction = (name, value) => {
+    switch (name) {
+      case "firstName":
+        return setValidFirstName(nameRegex.test(value));
+      case "lastName":
+        return setValidLastName(nameRegex.test(value));
+      case "email":
+        return setValidEmail(emailRegex.test(value));
+      case "password":
+        return setValidPasswordFormat(passwordRegex.test(value));
+      case "confirmPassword":
+        const { password } = { ...userCredentials };
+        return setValidPassword(validatePassword(password, value));
+      default:
+        return null;
+    }
+  };
+  /**
+   * @description Handles on change events of the form. closes alert
+   * @param {Event} event
+   */
+  const onChange = event => {
+    const { name, value } = event.target;
+    getValidationFunction(name, value);
+    setUserCredential(prev => ({ ...prev, [name]: value }));
+    setAlertNotification(defaultError);
+  };
+  /**
+   * @description Check if condition is true or false. Shortcut to writting full condition.
+   * document.getElementById is used when the form is already submitted but an error happened.
+   * Since the confirmPassword is removed, it will no longer be in the userCredentials.
+   * @param {Boolean} boolean_value check if true or false.
+   * @param {Object} checkState the state to check against.
+   * @param {String} name the name of the input field.
+   */
+  const checkValidity = (boolean_value, checkState, name) => {
+    const inputField =
+      name === "confirmPassword"
+        ? document.getElementById("confirmPassword")
+        : userCredentials[[name]];
+    if (checkState === boolean_value && inputField && inputField !== "")
+      return true;
+    else {
+      return false;
+    }
+  };
+  /**
+   * @description Handles form submittion and checks validity of inputs.
+   * @param {Event} e The event after clicking on the button.
+   */
+  const handleButtonClick = async e => {
+    e.preventDefault();
+    if (
+      checkValidity(true, validPassword, "confirmPassword") &&
+      checkValidity(true, validPasswordFormat, "password") &&
+      checkValidity(true, validFirstName, "firstName") &&
+      checkValidity(true, validLastName, "lastName") &&
+      checkValidity(true, validEmail, "email")
+    ) {
+      delete userCredentials["confirmPassword"]; //remove confirm password
+      const response = await register(userCredentials);
+      if (response.statusCode === 201)
+        triggerAlert("success", "Registration Successful!");
+      //TODO should redirect this.props.history.push("/dashboard");!!!
```

Comment:
If i recall, we should already have react-router installed and setup. Just call

` this.props.history.push("/dashboard");` right away